source/usage-examples/find.txt

```diff
@@ -0,0 +1,69 @@
+=======================
+Find Multiple Documents
+=======================
+
+.. default-domain:: mongodb
+
+You can query for multiple documents in a collection by calling the ``find()``
+method on a ``MongoCollection`` object. Pass a query(TODO: link to filter builder) filter to the
+``find()`` method to query for and return documents that match the filter in
+the collection. If you do not include a filter, MongoDB returns all the
+documents in the collection.
+
+For more information on querying MongoDB with the Java driver, see our
+guide (TODO: link) to querying documents.
+
+You can also chain methods to the ``find()`` method such as ``sort()``
+(TODO: link) which organizes the matched documents in a specified order and
+``projection()`` (TODO: link) which configures the included fields in the
+return document.
+
+The ``find()`` method returns an instance of ``FindIterable``, a class
+that offers several methods to access, organize, and traverse the results.
+``FindIterable`` also inherits methods from its parent class,
+``MongoIterable`` which implements the core Java interface ``Iterable``.
+
+If you call the ``iterator()`` method on the ``MongoIterable``, it
+returns a ``MongoCursor`` which you can use to traverse the results.
+You can call methods on the ``MongoCursor`` such as ``hasNext()`` to check
+whether additional results exist, or ``next()`` to return the next document
+in the collection. If no documents match the query, calling ``hasNext()``
+returns ``false`` and therefore calling ``next()`` throws an exception.
+
+For additional information on the aforementioned classes and methods, see the
+following API documentation:
+
+- :java-sync-api:`FindIterable <com/mongodb/client/FindIterable.html>`
```

Comment:
It feels slightly awkward to me to have the links to the api documentation outside the flow of the text. They also brake the flow of the example by being in the middle of the text.  Would it be better to add the links to the API docs at the end?

Also a couple of concepts have not been documented / linked to in the API docs: Projection helpers (excludeId & include), Filters (lt) and Sorts (descending).

Reply:
Good idea, I'll relocate them to a section at the end and (eventually) see how that tests with users. I wonder if there should be a link instead to that section anchor so the users know where to look before jumping to google?

For context, each paragraph above the code example is organized as a logical section (sans headers), following the structure:

- Method and required parameter descriptions
- Optional parameter descriptions
- Return value without headers
- (optional) Common exceptions encountered and how to resolve them

Reply:
:+1: